website/project/model.py

```diff
@@ -3729,3 +3810,125 @@ def _on_reject(self, user, token):
             },
             auth=Auth(user),
         )
+
+
+class DraftRegistrationApproval(Sanction):
+
+    mode = Sanction.ANY
+
+    def _on_complete(self, user, token):
+        pass  # draft approval state gets loaded dynamically from this record
+
+    def _on_reject(self, user, token):
+        pass  # draft approval state gets loaded dynamically from this record
+
+
+class DraftRegistration(AddonModelMixin, StoredObject):
+
+    is_draft_registration = True
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    datetime_initiated = fields.DateTimeField(auto_now_add=True)
+    datetime_updated = fields.DateTimeField(auto_now=True)
+    # Original Node a draft registration is associated with
+    branched_from = fields.ForeignField('node')
+
+    initiator = fields.ForeignField('user')
+
+    # Dictionary field mapping question id to a question's comments and answer
+    # {
+    #   <qid>: {
+    #     'comments': [{
+    #       'user': {
+    #         'id': <uid>,
+    #         'name': <name>
+    #       },
+    #       value: <value>,
+    #       lastModified: <datetime>
+    #     }],
+    #     'value': <value>
+    #   }
+    # }
+    registration_metadata = fields.DictionaryField(default=dict)
```

Comment:
Will metadata only be used for comments and answers? (If so, is metadata the best name for this field?)